core/src/main/java/cucumber/runner/TestCaseSyncEventBus.java

```diff
@@ -0,0 +1,69 @@
+package cucumber.runner;
+
+import cucumber.api.event.Event;
+import cucumber.api.event.EventHandler;
+import cucumber.api.event.TestCaseEvent;
+import cucumber.api.event.TestCaseFinished;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TestCaseSyncEventBus implements EventBus {
+
+    // Static sync object so all instances can be locked together 
+    private static final Object SYNC = new Object();
```

Comment:
Global locks are bad for performance. For example if we create two instances of Runtime and execute them on different threads both  will have to contented to lock this guard. This isn't needed as neither Runtime should share any objects.

This can be avoided by implmeneting the `SynchronizedEventBus`. As it will allow us to lock on a instance of the bus associated with the `Runtime`.

Reply:
I mean to remove that, as originally in the code each RunnerSupplier was creating it’s own eventbus wrapper and they were flushing at the same time and causing formatters to receive events out of sync (separate tests)

Meant to remove it now that only one instance will be used

Reply:
Done